CONTRIBUTING.md

```diff
@@ -17,13 +17,314 @@
     under the License.
 -->
 
-## Contributing to Beam
+# Contributing to Beam
 
-*Before opening a pull request*, review the
-[Beam contribution guide](https://beam.apache.org/contribute/).
-It lists steps that are required before creating a PR. In particular, consider
+There are many ways to contribute to Beam, just one of which is by contributing code.
+For a full list of ways to contribute and get plugged into Beam, see the
+[Beam Contribution Guide](https://beam.apache.org/contribute/get-started-contributing/)
+
+## Code Contributions
+
+*Before opening a pull request*, review the Beam contribution guide below.
+It lists steps that are required before creating a PR and provides tips for
+getting started. In particular, consider the following:
 
 - Have you searched for existing, related Issues and pull requests?
 - Have you shared your intent by creating an issue and commenting that you plan to take it on?
 - If the change is large, have you discussed it on the dev@ mailing list?
 - Is the change being proposed clearly explained and motivated?
+
+These steps and instructions on getting started are outlined below as well.
+
+### Prerequisites
+
+- A [GitHub](https://github.com/) account.
+- A Linux, macOS, or Microsoft Windows development environment.
+- Java JDK 8 installed.
+- [Docker](https://www.docker.com/) installed for some tasks including building worker containers and testing changes to this website locally.
+- For SDK Development:
+  - [Go](https://golang.org) 1.16.0 or later installed for Go SDK development.
```

Comment:
I remember Go has to be installed even for Java SDK development.

Reply:
Moved it out of SDK development and removed `for Go SDK development`